TechnicalDocuments/Proposals/BuildersToFunctions.md

```diff
@@ -0,0 +1,161 @@
+    }
+
+    func handle(_ route: AppointmentListViewModel.Route) {
+        switch route {
+        case let .showAlert(error):
+            Builder.Alert.makeAlert(for: error, dismiss: secondary.dismiss)
+                |> secondary.present
+        }
+    }
+}
+```
+
+I will provide further examples in the proposal
+
+## Motivation
+
+There are a couple of reasons for this change:
+
+1. It was decided that DI via `Current` would happen at the builders level. This means that the requirement to inject dependencies via initializer, in builders, cease to exist. This would remove: protocols, clean FlowControllers APIs and dissolve the concept of child builders. 
+2. It would become easier to find factory methods to construct screens. One could just type `Builder.` and Xcode would autocomplete with the available factory methods. It would also free engineers from wasting too much time finding dependencies if they are immediately accessible via `Current` and the namespaced free functions. This is not a justification for engineers to not understand the codebase and how to do things "properly". It is simply an observation of how cumbersome passing dependencies via initializer is, the raison d'être for `Current` and the overall way of creating screens (VM + VC + FC). 
+
+The first point is about simplicity and clean-up of multiple ways of creating screens, while the second point focuses on discoverability and development ergonomics.   
+
+## Proposed solution
+
+**Note:** The proposed solution assumes that a way to authenticate HTTP requests is available at `Current` level (in this case something in the shape of `MainUserSession` abstracted via `UserSession`).
+
+Let's use `MedicalHistoryBuilder.swift` as an example, since it offers a trivial API: 
+
+``` swift
+init(session: UserSession, visuals: VisualDependenciesProtocol)
+```
+
+And the corresponding public API:
+
+```swift
+func make(modal: Flow) -> UIViewController
+```
+
+Given the Module where it's located it could be migrated to:
+
+```swift
+extension Builder {
+    enum ClinicalRecords {
+        func makeMedicalHistory(modal: Flow) -> UIViewController
+    }
+}
+```
+
+Particular medical histories (e.g. for minors) can also be created, but in this case, the minor session would be passed instead of relying on `Current`. This situation (with child builders) can be observed at `ClinicalRecordsChildBuilders.swift`:
+
+```swift
+extension Builder {
+    enum ClinicalRecords {
+        func makeMedicalHistory(with session: Session, modal: Flow) -> UIViewController
+        func makeMedicalHistory(modal: Flow) -> UIViewController
+    }
+}
+```
+
+This would remove the intricated structure we have with builders and its children.
+
+A side effect of this approach, because we are dealing with functions (versus structs) we are able to remove local state. An example of such case is `enableClinicalDownloads` at `ClinicalRecordsChildBuilders`. You can see the code smell here:
```

Comment:
```suggestion
A side effect of this approach, because we are dealing with functions (versus structs) we are able to remove local state. An example of this is `enableClinicalDownloads` in `ClinicalRecordsChildBuilders`. You can see the code smell here:
```